elytron/WFCORE-4360-CredentialStore_Expression_Resolution.adoc

```diff
@@ -0,0 +1,284 @@
+ * `data` - The clear text value to be encrypted.
+
+The return value of the operation will be the complete expression ready to be used elsewhere in the mode.
+
+We will not support the decryption of expressions other than the support within the management model to resolve existing expressions.
+
+
+
+==== Base64 Representations
+
+Base64 representations will be used for both the export and import of the generated keys and for the expression value to be passed to the expression resolver - generally these details should be opaque to the end user however we may include the description within the documentation allowing for others to make use of the format.  We should consider that in the future alternative representations may be required either to support alternative key types or to support alternative representations so the representations will also be versioned.
+
+The general structure of the first representation will be (illustrated as an array): -
+
+    { 'E', 'L', 'Y', (VERSION), (TYPE), VALUE... }
+
+For the changes being made for this enhancement the version will be `1`.  For each of the supported commands and operations should a representation with a version other than `1` an error will be reported. 
+
+A generated `SecretKey` will be encoded with a type of 1: -
+
+    { 'E', 'L', 'Y', (VERSION), 1, KEY... }
+
+As version `1` of this implementation only supports `AES` `SecretKey` instances it is not necessary for the key algorithm to be encoded within this representation, additionally the key size does not need to be specified as it will be detectable from the length of the actual key.
+
+An encrypted expression will be encoded with a type of 2: -
+
+    { 'E', 'L', 'Y', (VERSION), 2, IV Length, Initialisation Vector, Cipher Text }
+
+At the moment only a single version of the representation will be supported, in the future the commands and operations that generated the encoded values may take a parameter to specify which version to generate however that is not required within the first implementation.  Additionally later versions of the tooling and operations should always attempt to use the oldest version possible to encode a value allowing newer tools to encode values for use by older versions.
+
+These representations are independent of the overall representation of the expressions as it is intended a portion of those expressions is both human readable and modifiable to allow alternative resolver definitions to be used without requiring the Base64 representation to be recreated. 
+
+==== Domain Mode
+
+The domain mode relationships will need to considered further, where we used the PicketBox vault we defined the vault within the host.xml and used it's values in the domain.xml.  As the credential store can both be defined in the host.xml and the domain.xml it would make sense to make use of the one stored within that descriptor, however this could be impacted based on how the configurable expression resolver is defined.
+
+
```

Comment:
@darranl This section should include discussion of how the various runtime-only management operations discussed earlier (i.e. manipulating the keystore) will be supported in domain mode; i.e. whether they will be available on /profile=foo/subsystem=elytron or whether the more typical behavior of only being available at /host=x/server=y/subsystem=elytron will be followed.